metadata-io/src/main/java/com/linkedin/metadata/search/utils/ESUtils.java

```diff
@@ -169,6 +170,7 @@ public static BoolQueryBuilder buildFilterQuery(
                 }
               });
       finalQueryBuilder.should(andQueryBuilder);
+      finalQueryBuilder.minimumShouldMatch(1);
```

Comment:
If the bool query includes at least one should clause and no must or filter clauses, the default value is 1. Otherwise, the default value is 0.

When access controls are present, the default changes from 1 to 0 otherwise.